### PR TITLE
docs: update pagination style tab

### LINKED ISSUE #4021: [Pagination, PaginationNav] update Style docs
## Acceptance criteria

- [x] Update the pagination component Style tab documentation with the new [content template](https://carbondesignsystem.com/contributing/documentation/#style-template).
- [x] Add pagination nav as another variant.
- [ ] Review
---

**Specs for reference**

Pagination nav

### PR DESCRIPTION
**Pull request by laurenmrice**
Closes #4021  

- This PR updates the pagination component Style tab documentation with using the [content template](https://carbondesignsystem.com/contributing/documentation/#style-template).
- It adds in pagination nav as another variant.

---

#### Changelog

**New**
- Added the pagination nav variant
- Added many sections throughout the doc

**Changed**
- Content and images

**Removed**
- Content and images that are no longer relevant

### CODE REVIEW

**Comment by laurenmrice** [comment]
Hey @emyarod or @alisonjoseph , when one of you get a chance can you see why my vercel deployment keeps failing? I can't seem to figure it out. Thanks!

**Comment by emyarod** [comment]
@laurenmrice it looks like `pagination-usage-sizes.png` is missing from the repo!

File: src/pages/components/pagination/style.mdx [hunk 1]
```diff
@@ -69,3 +182,33 @@ it's being paired with.
 | Icon button      | Small (sm)  | 32 / 2          |
 |                  | Medium (md) | 40 / 2.5        |
 |                  | Large (lg)  | 48 / 3          |
+
+#### Pagination size
+
+<Row>
+<Column colLg={12}>
+
+![Small, medium, and large sizes of pagination](images/pagination-usage-sizes.png)
```
Comment (alisonjoseph) [inline src/pages/components/pagination/style.mdx:191]: @laurenmrice this image is missing

Comment (laurenmrice) [reply src/pages/components/pagination/style.mdx:191]: Oh wow! Thanks

**Review by alisonjoseph — APPROVED** [review]

File: src/pages/components/pagination/style.mdx [hunk 2]
```diff
@@ -1,65 +1,190 @@
...
 <Caption>Spacing measurements for pagination | px / rem</Caption>
 
+#### Pagination nav structure
+
+<br />
+
 <div className="image--fixed">
 
-![Structure measurements for pagination](images/pagination-style-2.png)
+![Structure measurements for pagination nav](images/pagination-style-structure-pagination-nav.png)
```
Comment (thyhmdo) [inline src/pages/components/pagination/style.mdx:178]: This image is broken

File: src/pages/components/pagination/style.mdx [hunk 3]
```diff
@@ -1,65 +1,190 @@
...
-| Container      | border                      | 1px       | –             |
-|                | padding-left, padding-right | 16 / 1    | `$spacing-05` |
-| Select control | padding-left, padding-right | 16 / 1    | `$spacing-05` |
-| Arrow icon     | size                        | 20 x 20px | –             |
+<br />
 
 <div className="image--fixed">
 
-![Spacing measurements for pagination](images/pagination-style-1.png)
+![Spacing measurements for pagination](images/pagination-style-structure-pagination.png)
```
Comment (thyhmdo) [inline src/pages/components/pagination/style.mdx:166]: This image is broken

File: src/pages/components/pagination/style.mdx [hunk 4]
```diff
@@ -1,65 +1,190 @@
...
-<Caption>Structure measurements for pagination | px / rem</Caption>
+<Caption>Structure measurements for pagination nav | px / rem</Caption>
+
+## Size
 
-## Sizes
+The pagination component is available in three sizes for both variants:
+**large**, **medium**, and **small**.
 
 | Element          | Size        | Height (px/rem) |
```
Comment (thyhmdo) [inline src/pages/components/pagination/style.mdx:189]:
```suggestion
| Element          | Size        | Height (px / rem) |
```

File: src/pages/components/pagination/style.mdx [hunk 5]
```diff
@@ -69,3 +194,37 @@ it's being paired with.
...
+<Caption>Small, medium, and large sizes of pagination</Caption>
+
+#### Pagination nav size
+
+<br />
+
+<Row>
+<Column colLg={12}>
+
+![Small, medium, and large sizes of pagination nav](images/pagination-usage-sizes-pagination-nav.png)
```
Comment (thyhmdo) [inline src/pages/components/pagination/style.mdx:219]: In the last 2 rows, the selected number 4 looks like it's `body-compact`?

File: src/pages/components/pagination/style.mdx [hunk 6]
```diff
@@ -1,65 +1,190 @@
...
+<Caption>Example of pagination interactive state color</Caption>
+
+#### Pagination nav interactive state color
+
+<br />
+
+<Row>
+<Column colLg={12}>
+
+![Example of pagination nav interactive state color](images/pagination-style-interactive-pagination-nav.png)
```
Comment (thyhmdo) [inline src/pages/components/pagination/style.mdx:122]: In the last 2 rows, the selected number 4 looks like it's body-compact?

File: src/pages/components/pagination/style.mdx (hunk 5 above)
Comment (thyhmdo) [reply src/pages/components/pagination/style.mdx:219]: Visually, not sure why it looks heavier to the right and not centered. 

image

**Review by thyhmdo — APPROVED** [review]
Looks great!

**Pull request merged**